### ISSUE TITLE
sst deploy fails after Cognito user pool schema changes are wrongly detected

### ISSUE THREAD

**Issue by paulswail** [sst/ion#1319]
I've hit a blocker issue with `sst deploy` and `sst diff` detecting that my Cognito user pool's schema field has changed and it then attempts to deploy it and fails with the error "cannot modify or remove schema items: provider=aws@6.56.1", which is to be expected given that Cognito doesn't allow this.
However, these fields have not changed in my code or in my AWS account and had been working fine through several deploys until I ran `sst refresh` when attempting to resolve a separate issue (Neon provider plugin crashed).  From then onwards, I get this error when I run deploy.

I'm using SST v3.2.36.

Here's my user pool definition:

```ts
export const mainUserPool = new sst.aws.CognitoUserPool("MainPool", {
  aliases: ["email"],
  transform: {
    userPool: {
      schemas: [
        {
          name: "user_id",
          attributeDataType: "String",
          developerOnlyAttribute: false,
          mutable: true,
          required: false,
        },
      ],
    },
  },
});

```

When I run `sst diff`, I get: 

```
*  MainPool sst:aws:CognitoUserPool → MainPoolUserPool aws:cognito:UserPool
   * schemas[0].attributeDataType = String
   + schemas[0].developerOnlyAttribute = false
   * schemas[0].mutable = true
   * schemas[0].name = user_id
   + schemas[0].required = false
   * schemas[0].stringAttributeConstraints
```

**Comment by paulswail** [comment]
Would be nice if the `sst diff` command had a `--verbose` argument to show for `*` items that have changed, what was the value it's being changed from so I could see why it thinks this field has changed

**Comment by jayair** [comment]
That's strange that it would detect that something changed when there were no changes. Could it be possible that the AWS provider was updated in between? Or did this happen between deploys without any changes in between?

**Comment by paulswail** [comment]
Re: AWS provider bring updated, I didn’t explicitly run sst install. Is there another way this could have been updated?

This first happened yesterday and I ended up just deleting the whole stack to get unblocked. I redeployed new stack today and it had been deploying fine through multiple deploys until I ran sst refresh.

Referencing this Discord thread where another user hit the same issue: https://discord.com/channels/983865673656705025/983865673656705028/1290605674656694296

**Comment by paulswail** [comment]
Adding some more info.
Running `sst state edit` to open the local state file, this is what is in the `outputs` of my user pool resource: 

```json
"schemas": [
                            {
                                "attributeDataType": "String",
                                "developerOnlyAttribute": false,
                                "mutable": true,
                                "name": "user_id",
                                "numberAttributeConstraints": null,
                                "required": false,
                                "stringAttributeConstraints": {
                                    "maxLength": null,
                                    "minLength": null
                                }
                            }
                        ],
```

And when I use `aws cognito-idp describe-user-pool --user-pool-id XXXX --query 'UserPool.SchemaAttributes'`, this is what I get:

```json
{
        "Name": "custom:user_id",
        "AttributeDataType": "String",
        "DeveloperOnlyAttribute": false,
        "Mutable": true,
        "Required": false,
        "StringAttributeConstraints": {}
    }
```

I can't spot anything that the diff command would flag as having changed.
